pipelines/build/common/build_base_file.groovy

```diff
@@ -124,12 +124,25 @@ class Builder implements Serializable {
     }
 
     List<String> getTestList(Map<String, ?> configuration) {
-        if (configuration.containsKey("test")) {
+        List<String> nightly = ['sanity.openjdk', 'sanity.system', 'extended.system', 'sanity.perf', 'sanity.external']
+        List<String> weekly = [
+                               'extended.openjdk', 'extended.perf', 'extended.external',
+                               'special.openjdk','special.functional', 'special.system', 'special.perf'
+                              ]
+        if (configuration.containsKey("test") && configuration.get("test")) {
             def testJobType = release ? "release" : "nightly"
             if (isMap(configuration.test)) {
-                return (configuration.test as Map).get(testJobType) as List<String>
+                if ( testJobType == "nightly" ) {
+                    return (configuration.test as Map).get("nightly") as List<String>
+                } else {
+                    return ((configuration.test as Map).get("nightly") as List<String>) + (configuration.test as Map).get("weekly") as List<String>
```

Comment:
Long return statement, prefer splitting it out into individual statements with a simpler return at the end

Reply:
yes, updated.